contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -1085,6 +1085,44 @@ export const Channel = new Resource({
     });
   },
 
+  deploy(id) {
+    const change = {
+      key: id,
+      source: CLIENTID,
+      table: this.tableName,
+      type: CHANGE_TYPES.DEPLOYED,
+      channel_id: id,
+    };
+    return this.transaction({ mode: 'rw', source: IGNORED_SOURCE }, CHANGES_TABLE, () => {
+      return db[CHANGES_TABLE].put(change);
+    });
+  },
+
+  waitForDeploying(id) {
+    const observable = Dexie.liveQuery(() => {
+      return this.table
+        .where('id')
+        .equals(id)
+        .filter(channel => !channel['staging_root_id'] || channel['staging_root_id'] === null)
```

Comment:
@vkWeb `!channel['staging_root_id']` should be `true` when `channel['staging_root_id'] === null` is true

Reply:
@bjester sir, actually after lots of manual testing I observed that sometimes the `staging_root_id` was getting removed as a property and sometimes it was being set as `null`. So to handle the case when `staging_root_id` is not a property of `channel` resource I'm using `!channel['staging_root_id']`.